chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -708,7 +678,8 @@ async def generate_unsigned_spendbundle(
             elif len(payments) > 1:
                 raise ValueError("NFTs can only be sent to one party")
 
-            nft_coin = next(c for c in self.my_nft_coins if c.coin in coins)
+            nft_coin = await self.nft_store.get_nft_by_coin_ids([x.name() for x in coins])
```

Comment:
> It also seems a bit misleading to make coins a Set, since it's only allowed to contain a single coin at most.
we're using `coins` to stay compatible with the interface of `generated_unsigned_bundle` that all other wallets use